### DISCUSSION TITLE
Full AoT: running dynamically loaded assemblies

### DISCUSSION THREAD

**Discussion by andersson09** [unoplatform/Uno.Wasm.Bootstrap#379] [Q&A]
This is more of question. On the main page it states

"This mode will not allow the execution of code that was not known at compile time (e.g. dynamically generated assemblies or loaded through Assembly.LoadFrom)."

Whats the particular reason for this. Is this something which is impossible to achieve with Full AoT or just not implemented yet?

Thanks

**Comment by jeromelaban** [comment]
This is how Full AOT works, this is a mode that was primarily designed to work on non-JIT allowing environments like iOS. This mode cannot accept any unknow binary because it does not know how to execute IL (there's no interpreter).

That being said, "Mixed Mode" (or `InterpreterAndAOT`) allows for arbitrary code to be executed without any issues, this is something I'm using in [SkiaSharp Fiddle](https://github.com/jeromelaban/uno.skiasharpfiddle), which compiles C# code in memory and runs it through the interpreter ([see for yourself](https://lemon-ocean-02b7cb210.azurestaticapps.net/)).

**Reply by andersson09** [reply]
Thanks that makes sense. However what would then be the advantage of Full AoT over mixed mode when simply executing code known at compile time?

**Reply by jeromelaban** [reply]
Over time, the advantages of the Full AOT mode have faded. It used to be useful because the "Mixed Mode" did not exist. 

Moreover, currently, FullAOT is not functioning properly with .NET 6 because there are missing WebAssembly features (e.g. exceptions) which prevent some specific language features from working properly.
